trainer/metrics.py

```diff
@@ -82,4 +85,117 @@ def save_metrics(task_name, model_dir):
   metrics_path = os.path.join(
       model_dir,
       "validation_eval/metrics" + str(results["checkpoint_step"]) + ".json")
-  json.dump(scores, tf.io.gfile.GFile(metrics_path, "w"))
+  json.dump({"nltk_bleu_score": nltk_bs, "sacrebleu_blue_score": sb_bs,
+             "recall@1": 0}, tf.io.gfile.GFile(metrics_path, "w"))
+
+def sklearn_recall(targets, predictions):
+  """Uses the built in t5 sklearn_metrics_wrapper to calculate sklearn recall@1
+
+  Args:
+    targets: a list of strings, the target from the validation set
+    preditcions: a list of strings, the model predictions
+
+  Returns:
+    a dictionary: {"sklearn_recall": recall_value}
+  """
+  prfs = t5.evaluation.metrics.sklearn_metrics_wrapper(
+      "precision_recall_fscore_support",
+      average='micro')(targets, predictions) \
+      ["precision_recall_fscore_support"]
+  return {"sklearn_recall": prfs[1]}
+
+def t2t_bleu(targets, predictions):
+  """Tokenizes with the bleu_tokenize method from the t2t library then
+  calls the compute_bleu function
+
+  Args:
+    targets: a list of strings, the target from the validation set
+    preditcions: a list of strings, the model predictions
+
+  Returns:
+    a dictionary: {"t2t_bleu": bleu_value}
+  """
+  targets_tokens = [bleu_hook.bleu_tokenize(x) for x in targets]
+  predictions_tokens = [bleu_hook.bleu_tokenize(x) for x in predictions]
+  return {"t2t_bleu": 100 * bleu_hook.compute_bleu(targets_tokens,
+                                                   predictions_tokens)}
+
+def replace_titles(targets, predictions):
+  """Replaces titles with a __unk__ token. Returns an updated
+  (targets, predictions) tuple"""
+  replace_fn = \
+    lambda text: re.sub(r"\@([^@]*\([^@]*\)[^@]*)\@", "__unk__", text)
+  return (list(map(replace_fn, targets)), list(map(replace_fn, predictions)))
+
+def bleu_no_titles(targets, predictions):
+  """Bleu metric with titles removed.
```

Comment:
Probably useful to mention the "why" as to why needed compared to the standard bleu?